test/unit/associations/association.test.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+/* jshint -W030 */
+const chai = require('chai');
+const expect = chai.expect;
+const Support = require(__dirname + '/../support');
+const current = Support.sequelize;
+const AssociationError = require(__dirname + '/../../../lib/errors').AssociationError;
+
+describe(Support.getTestDialectTeaser('belongsTo'), function() {
+  it('should throw an AssociationError when two associations have the same alias', function(){
+    const User = current.define('User');
+    const Task = current.define('Task');
+
+    User.belongsTo(Task, { as: 'task' });
+    const errorFunction = User.belongsTo.bind(User, Task, { as: 'task' });
+    const errorMessage = `You have used the alias task in two separate associations. Aliased associations must have unique aliases.`
```

Comment:
We have rudimentary support for ESlint based rules, but its not adopted fully in our codebase. Progress on complying full codebase to ESlint rules is bit slow. In future we want to use ESlint to catch all these errors automatically :)